File: 30_days_C#_HackerRank_Challenge/RegExPatternDB.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {
    static void Main(String[] args) {
        int N = Convert.ToInt32(Console.ReadLine());
        List<string> names = new List<string>();

        for (int i = 0; i < N; i++) {
            string[] inputs = Console.ReadLine().Split(' ');
            string firstName = inputs[0];
            string emailID = inputs[1];

            if (emailID.EndsWith("@gmail.com")) {
                names.Add(firstName);
            }
        }

        names.Sort();
        foreach (string name in names) {
            Console.WriteLine(name);
        }
    }
}

File: 30_days_Python_HackerRank_Challenge/RegExPatternDB.py
#py 2
#!/bin/python
import sys
n = int(raw_input().strip())
names = []
for _ in xrange(n):
    firstName, emailID = raw_input().strip().split()
    
    if emailID.endswith("@gmail.com"):
        names.append(firstName)

names.sort()
for name in names:
    print name

#py 3
#!/bin/python3
import sys
n = int(input().strip())
names = []
for _ in range(n):
    firstName, emailID = input().strip().split()
    
    if emailID.endswith("@gmail.com"):
        names.append(firstName)
names.sort()

for name in names:
    print(name)
